### ISSUE TITLE
Trying to prevent "FileNotFoundError: [Errno 44]"

### ISSUE THREAD

**Issue by pierrepo** [jupyterlite/jupyterlite#1371]
This issue is both a bug report and a documentation improvement, so I report it as a feature request :-)

### Problem

I'm using JupyterLite for teaching because it could be easily deployed on GitHub pages.

My students are sometimes reporting the error message `FileNotFoundError: [Errno 44] No such file or directory` documented in the [Troubleshooting](https://jupyterlite.readthedocs.io/en/stable/troubleshooting.html) section. Here is my attempt to better understand how and when this error occurs:

From all the tests I performed, it looks like this error could be reproduced:
- with Mozilla Firefox (both on Windows and Linux)
- when the user executes a Python code cell to access/read a file **before** the notebook is fully loaded. By this, I mean the Kernel status is *Unknown* 
![image](https://github.com/jupyterlite/jupyterlite/assets/733861/e5659073-9297-4293-b13a-225321085f9f)
instead of *Iddle*
![image](https://github.com/jupyterlite/jupyterlite/assets/733861/91043d3d-c31b-41e9-9e2b-a7373c144931)

Once the `FileNotFoundError: [Errno 44] No such file or directory` is obtained, reloading the page or the kernel, does not help. Files can't be read even if the notebook is properly loaded and the kernel status is *Iddle*. The only solution is then to clear the browser cache.

### Proposed Solution

1. Documentation-based solution: mention in the documentation that the notebook should be fully loaded (and the kernel Idlle) before trying to access any local files.

2. Code-based solution: prevent the user to perform any interaction with the notebook until the kernel is iddle. This could be an opt-in option has this solution may not fit all users.

**Comment by Carreau** [comment]
Thanks for the report, 

I never tried on Firefox for mac (124.0.2), but jupyterlite seem to not work for me:

```
Error
Repository is read only
```

Which is weird. 

On chrome (123.0.6312.107) I have no issues with executing, it will just wait until the kernel is ready.

But we can indeed add a troubleshooting section warning about this problem.

**Comment by ArturoAmorQ** [comment]
> From all the tests I performed, it looks like this error could be reproduced:
   with Mozilla Firefox (both on Windows and Linux)

This issue also seems to appear in chrome-based browsers.

**Comment by ogrisel** [comment]
I can trigger `FileNotFoundError` but only if I open the same jupyterlite site in 2 concurrent browser tabs as reported here: #1582. But I don't think it's related to executing cells when Kernel status is Unknown.

**Comment by pierrepo** [comment]
> @pierrepo do you have a public notebook that can trigger this bug easily?

@ogrisel I reported this bug while executing [this notebook](https://github.com/pierrepo/intro-bioinfo-python-2024/blob/main/content/5_fichiers.ipynb) from [this repo](https://github.com/pierrepo/intro-bioinfo-python-2024/) in my class.

You can run the corresponding JupyterLite env [here](https://pierrepo.github.io/intro-bioinfo-python-2024/lab/index.html).

Please note this repo has not been updated since April 11th, 2024 and sadly the bug did not occur systematically.

**Comment by ogrisel** [comment]
Thanks, @pierrepo. I cannot reproduce the `FileNotFoundError` error you describe after restarting the pyodide and executing the first cells manually more than 10 times in a row.

However, I can reproduce another bug that was fixed in more recent jupyterlite versions: #1464.

However, the latter never causes `FileNotFoundError` to be raised.

Did you it's possible that you or your students had several browser tabs open to https://pierrepo.github.io/intro-bioinfo-python-2024/lab/index.html at the same time whe you observed this? If so, this might be the same problem as #1582 which I can reproduce almost 100% of the time with the reproduction instructions I wrote there.

**Comment by pierrepo** [comment]
Hello @ogrisel thanks for investigating further. We indeed could have had several tabs open on the same JupyterLite site. But I'm not 100% sure about this.

**Comment by jtpio** [comment]
I'm also not able to reproduce on Firefox and Chrome on MacOS.

Also the execute requests should normally be queued until the kernel is ready. Maybe there is still some for improvements on that level though.

Thinking more about https://github.com/jupyterlite/jupyterlite/issues/1582, it should in theory be happening more often when using the Jupyter Notebook 7 interface, which opens notebooks in different browser tabs.

**Comment by ogrisel** [comment]
> Thinking more about https://github.com/jupyterlite/jupyterlite/issues/1582, it should in theory be happening more often when using the Jupyter Notebook 7 interface (/notebooks), which opens notebooks in different browser tabs.

Yes, this is precisely the setting where I observed the problem most often. I switched to the notebook 7 interface as a tentative mitigation for the #1527 and the #1359 issues (probably caused by https://github.com/pyodide/pyodide/issues/5264).
